internal/lib/avi_api.go

```diff
@@ -66,10 +66,10 @@ func AviGet(client *clients.AviClient, uri string, response interface{}, retryNu
 		if aviError, ok := err.(session.AviError); ok && aviError.HttpStatusCode == 403 {
 			utils.AviLog.Debugf("Switching to admin context from %s", GetTenant())
 			SetAdminTenant := session.SetTenant(GetAdminTenant())
-			SetTenant := session.SetTenant(GetTenant())
 			SetAdminTenant(client.AviSession)
+			SetTenant := session.SetTenant(GetTenant())
 			defer SetTenant(client.AviSession)
-			if err := AviGet(client, uri, response, retry+1); err != nil {
+			if err = AviGet(client, uri, response, retry+1); err != nil {
```

Comment:
not sure how this solves the issue. Can this be simulated using our UTs or FTs?

Reply:
err variable has to be of a bigger scope. 
Issue was once this block is finished this local `err` variable values get destroyed and `err` now contains values obtained from line 63 (rest call with non-admin tenant).  
IT can be simulated with FTs for  non-admin tenant and we can check if ingress has cluster-uuid  populated,

Reply:
I got it. Thanks. In that case, if the call with the admin tenant will return err as nil and should we proceed with the other checks in L77, L79, and all?

Reply:
added an else block to return cleanly.